napari/_vispy/layers/base.py

```diff
@@ -48,14 +49,17 @@ def __init__(self, layer, node):
         self.layer = layer
         self._array_like = False
         self.node = node
+        self.bounding_box_2D = BoundingBox(parent=self.node)
+        self.bounding_box_3D = BoundingBox(parent=self.node)
```

Comment:
Probably better, yeah. Unfortunately, the switching will still be necessary (at least for `Line`) inside that class, because `vispy` does not let us change the data of a `Line` from 2d to 3d once it was set once.